problems/coj_problems.go

```diff
@@ -1,139 +1,136 @@
-        log.Println("Parse response to buffer failed: ", errBuffer)
-        return sumbitionsInfo, errBuffer
-    }
-
-    // Note: The Judge response is a JSON array, we should parse each element as a COJJudgment
-    // so we can use it as a struct. We can use other approach that is parse the JSON as a map
-    // See the example below:
-    //      data := make([] map[string]interface{}, 0)
-
-    data := make([] COJJudgement, 0)
-    errParse := json.Unmarshal(buffer, &data)
-    if errParse != nil {
-        log.Println("Parse response to COJJudgement failed: ", errParse)
-        return sumbitionsInfo, errParse
-    }
-
-    // Iterate over all submitions to check if the problem has been solve by the user.
-    // We can make another call to the API adding an parameter status=ac but we already have all the
-    // submitions that is a waste of time.
-    for _, sumbition := range data {
-        if sumbitionsInfo.HasTried == false {
-            sumbitionsInfo.HasTried = true
-        }
-        if sumbition.Status == "Accepted" {
-            sumbitionsInfo.HasSolved = true
-        }
-    }
-    return sumbitionsInfo, nil
+func getHasSolvedInCOJ(pid string, username string) (SubmitionsProblemInfo, error) {
+	sumbitionsInfo := SubmitionsProblemInfo{false, false}
+
+	// Parameters sent to the API
+	apiParameters := map[string]string{"username": username, "pid": pid}
+	// Generate the URL for make the request
+	apiURL, err := generateRequestURL("GET", "http://coj.uci.cu/api/judgment", apiParameters)
+
+	if err != nil {
+		log.Println("GenerateResquestUrl failed: ", err)
+		return sumbitionsInfo, err
+	}
+
+	// Generate the http.Client to make the request. We do the request in this way so we can control
+	// the time out, otherwise we may have troubles if COJ API fails.
+	netClient := createNetClient()
+
+	response, errResponse := netClient.Get(apiURL)
+	if errResponse != nil {
+		log.Println("GET Resquest failed: ", errResponse)
+		return sumbitionsInfo, errResponse
+	}
+
+	buffer, errBuffer := ioutil.ReadAll(response.Body)
+	if errBuffer != nil {
+		log.Println("Parse response to buffer failed: ", errBuffer)
+		return sumbitionsInfo, errBuffer
+	}
+
+	// Note: The Judge response is a JSON array, we should parse each element as a COJJudgment
+	// so we can use it as a struct. We can use other approach that is parse the JSON as a map
+	// See the example below:
+	//      data := make([] map[string]interface{}, 0)
+
+	data := make([]COJJudgement, 0)
+	errParse := json.Unmarshal(buffer, &data)
+	if errParse != nil {
+		log.Println("Parse response to COJJudgement failed: ", errParse)
+		return sumbitionsInfo, errParse
+	}
+
+	// Iterate over all submitions to check if the problem has been solve by the user.
+	// We can make another call to the API adding an parameter status=ac but we already have all the
+	// submitions that is a waste of time.
+	for _, sumbition := range data {
+		if sumbitionsInfo.HasTried == false {
```

Comment:
Seems like this is a bug (which we didn't catch in the original review) it should be `submission.HasTried`, right?

Reply:
If a User X has at least a submission for a problem Y then he has automatically tried it.
It should not be there, I think it should be outside the loop.

Reply:
You know better, I just though that it might be a bug because in this first if you use `submitionsInfo` and in the second if you use `submition`. If you think it's correct then go for it.

Reply:
Fix in commit [571cbb0](https://github.com/algoristas/api/pull/14/commits/571cbb012145ab8efdf8650608cd7c82b4cbfc36)